libraries/SITL/SIM_TFS20L.cpp

```diff
@@ -0,0 +1,97 @@
+#include "SIM_TFS20L.h"
+#include "SITL/SIM_Aircraft.h"
+#include <AP_Math/AP_Math.h>
+#include <cmath>
+
+#if AP_SIM_TFS20L_ENABLED
+
+using namespace SITL;
+
+// Register addresses matching real sensor
+static const uint8_t REG_DIST_LOW = 0x00;
+static const uint8_t REG_DIST_HIGH = 0x01;
+static const uint8_t REG_AMP_LOW = 0x02;
+static const uint8_t REG_AMP_HIGH = 0x03;
+static const uint8_t REG_VERSION_MAJOR = 0x0C;
+
+// Limits matching real sensor
+static const uint16_t MAX_DIST_CM = 2000;  // 20m
+static const uint16_t MIN_DIST_CM = 1;     // 1cm
+
+void TFS20L::update(const class Aircraft &aircraft)  
+{
+    // Get range and convert to cm for checking limits
+    rangefinder_range = aircraft.rangefinder_range();
+    
+    // Check for invalid readings first (infinite or negative)
+    if (rangefinder_range <= 0 || rangefinder_range > (MAX_DIST_CM * 0.01f)) {
+        rangefinder_range = 0;  // Will trigger out of range
+        strength = 0;          // Low strength for invalid reading
+        return;
+    }
+    
+    const uint16_t range_cm = rangefinder_range * 100.0f;
+    
+    // Apply real sensor limits
+    if (range_cm > MAX_DIST_CM || range_cm < MIN_DIST_CM) {
```

Comment:
Should really be testing `range_cm` *before* throwing it into a `uint16_t as over 65m this will wrap around and be valid again :-)